### PR TITLE
Set up connection into postgres db docker

### PR DESCRIPTION
**Pull request by nuxeric**
Its a small PR, thought it would be easier to work in smaller steps 

I added a dockerfile for the backend project I thought that this would be easier to keep track of modularity instead of one big compose file, then I used the environment variables to set up the connection along with https://node-postgres.com/ package inside the backend project to set up the conneciton. 

I apologize for the really bad PR description I'll try to be more expressive but homework is really doing me in right now. Lmk if you have any questions.

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -14,10 +14,12 @@
   "license": "ISC",
   "dependencies": {
     "@types/dotenv": "^8.2.0",
+    "@types/pg": "^7.14.6",
     "@types/pino": "^6.3.1",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "helmet": "^4.1.1",
+    "pg": "^8.4.2",
```
Comment (pseigo) [inline backend/package.json:22]: One of us will have to rebuild our `package-lock.json` :D

**Review by pseigo — COMMENTED** [review]
I get the database connected message on my end. Any tips on seeing the website locally on Windows? I can't hit it from my browser or curl at localhost, 127.0.0.1, 0.0.0.0, or 172.19.0.3 (IP of VM) -- on :8080.

**Review by pseigo — CHANGES_REQUESTED** [review]
It'd be nice to have an example of making a DB query from a routed request (request -> router -> module that makes the DB call). It'd probably be easiest to do this on top of my changes (#2). If #2 is all good, we can merge that and you can rebase on those changes, or we could do it as a separate ticket.

Let me know what you'd like to do :)

File: docker-compose.yml [hunk 2]
```diff
@@ -13,5 +13,15 @@ services:
...
       POSTGRES_DB: cpsc304_dev
 
+  backend:
+    container_name: cpsc304-backend
+    build:
+      context: ./backend
+    depends_on:
+      - postgres
+    expose:
+      - "8080"
```
Comment (pseigo) [inline docker-compose.yml:23]: With this change I'm able to connect to the server via my WSL2's IP. I don't know anything about Docker though so maybe there's a better way? 🤷‍♂️

```suggestion
      - "8080"
    ports:
      - "8080:8080"
```

**Review by pseigo — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (nuxeric) [reply docker-compose.yml:23]: my bad, you're right 
difference between expose and ports below:
https://stackoverflow.com/questions/40801772/what-is-the-difference-between-docker-compose-ports-vs-expose

I forget that we are using the 'backend" to serve our frontend as well, we should change the naming convention to "web" instead for all related folders and files

Comment (pseigo) [reply docker-compose.yml:23]: @nuxeric Good idea! Created #4 to make that change quickly after these are merged

Comment (pseigo) [reply docker-compose.yml:23]: @nuxeric  Would this make `expose` redundant here then? And would it still be needed in the Dockerfile?

**Review by ShaunDonn — APPROVED** [review]
Looks like it's working!

**Comment by pseigo** [comment]
to-do:
1. https://github.com/nuxeric/cpsc304project/pull/1#discussion_r520951835
2. rebase on master

```bash
git fetch --all
git checkout feature/postgres-connection
git rebase origin/master
git status # anything unstaged has conflicts => go fix those
git add ... # conflicted files that you've fixed
git rebase --continue
git push
```

**Review by pseigo — APPROVED** [review]

**Pull request merged**